### PR TITLE
DB re-design, Model testing, endpoints and their endpoints

### PR DESCRIPTION
**Pull request by stiehlrod**
The problem I encountered was redesigning the database in order to efficiently make calls in two directions:
1. questions and answers
2. questions, answers and responses

We couldn't get the second option without an n+1 query. 

Stack Overflow has a high traffic discussion with a solution called delegation. Since finding the solution, I have reworked the database and created new endpoints and tests to make sure this works, and it works WELL.

Here is the SO post: https://stackoverflow.com/questions/4021322/belongs-to-through-associations

This was an interesting problem because at first glance it seems like there is an obvious solution, however on closer inspection it turns out to be a lot more complicated, especially when considering database performance.

This PR includes a new db schema, new relationships, including delegation, reconfigured endpoints and corresponding tests. The endpoints include:
1. `/api/v1/questions`  sends a get request and returns all the questions
2. `/api/v1/q_and_a`  sends a get request and returns the questions and possible choices
3. `/api/v1/responses` send a get request and returns the question_id, course_id, student_id, choice_id, answer_type, answer_text and timestamps.

These can be reconfigured later and I felt it was important to get the new design up and ready for @tnodland to containerize.

### CODE REVIEW

File: app/controllers/surveys.rb [hunk 1]
```diff
@@ -2,11 +2,46 @@
...
+  end
+
+  get '/api/v1/q_and_a' do
+    question_id = params[:question_id]
+    q_and_a = Question.get_q_and_a(question_id)
+    q_and_a.to_json
+  end
+
+  post '/api/v1/responses' do
+    response_data = params
```
Comment (wipegup) [inline app/controllers/surveys.rb:31]: Does this line collect info from both the headers and params?

I'm unfamiliar with the `sinatra` way this gets done.

**Review by wipegup — APPROVED** [review]
Everything looks good with the `get` requests!

File: app/controllers/surveys.rb (hunk 1 above)
Comment (stiehlrod) [reply app/controllers/surveys.rb:31]: forgive me, that is sudo code, I have not yet gotten back to the `post`.

**Comment by carriewalsh** [comment]
You did a great job with this Jennica!

**Pull request merged**